### PR TITLE
[DO NOT MERGE] Add documentation man page for security-secrets-setup.

### PR DESCRIPTION
**Pull request by bnevis-i**
Jim, can you please review for accuracy?

### CODE REVIEW

**Comment by bnevis-i** [comment]
This PR selected wrong file for review.  Will resubmit.

File: security/security-secrets-setup.1.rst [hunk 1]
```diff
@@ -0,0 +1,189 @@
...
+
+SYNOPSIS
+========
+
+| security-secrets-setup -config params.json
+| security-secrets-setup -generate
+| security-secrets-setup -cache
+| security-secrets-setup -import
+| security-secrets-setup [-h|--help]
+
```
Comment (jim-wang-intel) [inline security/security-secrets-setup.1.rst]: @bnevis-i Bryon, there is `-configDir` option mentioned below but not listed here.  Should that one is also listed here?

File: security/security-secrets-setup.1.rst [hunk 2]
```diff
@@ -0,0 +1,189 @@
...
+security-secrets-setup supports several modes of operation as defined in the OPTIONS section.
+
+As the PKI is security-sensitive,
+this tool takes a number of precautions to safeguard the PKI:
+
+* The PKI can be deployed to transient storage to address potential attacks to the PKI at-rest.
+
+* The PKI is deployed such that each service has its own assets folder, which is amenable to security controls imposed by container runtimes such as mandatory access controls or file system namespaces.
+
+* The CA private key is shred (securely erased) prior to caching or deployment to block issuance of new CA descendants (this is most relevant in caching mode).
```
Comment (jim-wang-intel) [inline security/security-secrets-setup.1.rst]: You might want to spell out what "CA" is.

File: security/security-secrets-setup.1.rst [hunk 3]
```diff
@@ -0,0 +1,189 @@
...
+In ``-generate`` or ``-cache`` modes,
+the utility hard-codes the names of the configuration files
+and always processes ``edgex-vault.json`` first
+and ``edgex-kong.json`` second.
+This hard-coding is done for now in order to meet the October Fuji deadline.
+
+
+configuration.toml
+------------------
+Configuration file for options.
```
Comment (jim-wang-intel) [inline security/security-secrets-setup.1.rst]: maybe state as ```Configuration file for configureable directories that options use```

File: security/security-secrets-setup.1.rst [hunk 4]
```diff
@@ -0,0 +1,189 @@
...
+New behavior: As security-secrets-setup is a helper utility to ensure that a PKI is
+created on first launch, 
+It is intended that security-secrets-setup is always invoked with the same operation flag,
+Changing from ``-cache`` to ``-generate`` will cause the cache to be ignored
+when deploying a PKI and changing it back will cause a reversion to a stale CA.
+Changing from ``-cache`` to ``-import`` mode of operation is not noticeable by the tool:
+the PKI that is in the cache will be the one deployed.
+To force regeneration of the PKI cache after the first launch,
+the PKI cache must be manually cleaned.
+The easiest way in Docker would be to delete the Docker volume holding the cached PKI.
```
Comment (jim-wang-intel) [inline security/security-secrets-setup.1.rst]: Can we make the key-takeaway notes as bullet points or at least separate paragraphs for easier reading purpose?

File: security/security-secrets-setup.1.rst [hunk 5]
```diff
@@ -0,0 +1,189 @@
...
+~~~~~~~~~
+(Used in generate, cache, and import modes.)
+Points to the base directory for the final deployment location of the PKI.
+If not specified, defaults to ``/run/edgex/secrets/pki/``
+where each ``service-name.json`` processed causes assets to be placed in
+``{DeployDir}/service-name``. 
+
+* ``server.crt`` for a PEM-encoded end-entity TLS certificate and
+* ``server.key`` for the corresponding private key
+* ``.security-secrets-setup.complete`` is a sentinel file created after assets are deployed
```
Comment (jim-wang-intel) [inline security/security-secrets-setup.1.rst]: Right now the sentinel file is still called .pki-init.complete.  I will change the code to reflect this.

File: security/security-secrets-setup.1.rst [hunk 6]
```diff
@@ -0,0 +1,189 @@
...
+
+
+NOTES
+=====
+
+For backward compatible behavior, use the ``-config`` option.
+
+New behavior: As security-secrets-setup is a helper utility to ensure that a PKI is
+created on first launch, 
+It is intended that security-secrets-setup is always invoked with the same operation flag,
```
Comment (eno-intel) [inline security/security-secrets-setup.1.rst]: The capitalization and punctuation makes it difficult to read. Correction below
"...launch, **it** is intended... **flag.** Changing from -cache to -generate..."

File: security/security-secrets-setup.1.rst [hunk 7]
```diff
@@ -0,0 +1,189 @@
...
+=====
+
+edgex-vault.json, edgex-kong.json
+---------------------------------
+
+(Used in generate, cache, and import modes.)
+Configuration files for certificate parameters.
+The basename of this file creates a corresponding directory under *DeployDir*.
+For example, ``edgex-vault.json`` would create assets under
+``/run/edgex/secrets/pki/edgex-vault/``.
```
Comment (eno-intel) [inline security/security-secrets-setup.1.rst]: DeployDir isn't introduced in this doc yet.
`/run/edgex/secrets/pki/`` seems like an arbitrary folder to use. I could see someone asking "why this folder?"

Comment (jim-wang-intel) [reply security/security-secrets-setup.1.rst]: @eno-intel isn't the line above saying so already about the `DeployDir`?  Also, this note is also about an example only I think.

Comment (eno-intel) [reply security/security-secrets-setup.1.rst]: Yes, but it is assuming the reader knows what DeployDir is and where it is located. If we assume this, then we don't need the explanation for DeployDir at the bottom of the file.

I would re-phrase this sentence as such...
For example, if using the default DeployDir ``/run/edgex/secrets/pki/``,  ``edgex-vault.json`` would create assets under
``/run/edgex/secrets/pki/edgex-vault/``.

Comment (bnevis-i) [reply security/security-secrets-setup.1.rst]: Fixed.

**Pull request closed without merging**